addons/mail/static/src/messaging/entity/notification_group/notification_group.js

```diff
@@ -0,0 +1,109 @@
+const { attr, one2many } = require('mail.messaging.EntityField');
+
+function NotificationGroupFactory({ Entity }) {
+
+    class NotificationGroup extends Entity {
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * Opens the view that allows to cancel all notifications of the group.
+         */
+        openCancelAction() {
+            const device = this.env.messaging.device;
+            if (this.notification_type === 'email') {
+                this.env.do_action('mail.mail_resend_cancel_action', {
+                    additional_context: {
+                        default_model: this.model,
+                        unread_counter: this.notifications.length,
+                    },
+                });
+            }
+            if (!device.isMobile) {
+                this.env.messaging.messagingMenu.close();
+            }
+        }
+
+        /**
+         * Opens the view that displays either the single record of the group or
+         * all the records in the group.
+         */
+        openDocuments() {
+            if (this.res_id) {
+                if (this.model === 'mail.channel') {
+                    const channel = this.env.entities.Thread.insert({
+                        id: this.res_id,
+                        model: this.model,
+                    });
+                    channel.open();
+                } else {
+                    this.env.messaging.openDocument({
+                        id: this.res_id,
+                        model: this.model,
+                    });
+                }
+            } else {
+                this._openDocuments();
+            }
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @override
+         */
+        _createInstanceLocalId(data) {
+            return `${this.constructor.entityName}_${data.id}`;
+        }
+
+        /**
+         * Opens the view that displays all the records of the group.
+         *
+         * @private
+         */
+        _openDocuments() {
+            const device = this.env.messaging.device;
+            if (this.notification_type === 'email') {
+                this.env.do_action({
+                    name: this.env._t("Mail Failures"),
+                    type: 'ir.actions.act_window',
+                    view_mode: 'kanban,list,form',
+                    views: [[false, 'kanban'], [false, 'list'], [false, 'form']],
+                    target: 'current',
+                    res_model: this.model,
+                    domain: [['message_has_error', '=', true]],
+                });
+            }
+            if (!device.isMobile) {
+                this.env.messaging.messagingMenu.close();
+            }
```

Comment:
I'm actually wondering if this shouldn't always become flux specific? This only makes sense when clicking on a messaging menu element?
If I have the messaging menu open and I'm opening a document from another mean, I don't really want my messaging menu closed?

Reply:
Or it's getting closed automatically anyway due to global click handler.

Reply:
I think you can remove this code, and let global click handler auto-close it.